c2rust-analyze/src/panic_detail.rs

```diff
@@ -136,3 +147,24 @@ fn panic_to_string(e: &(dyn Any + Send + 'static)) -> String {
 
     format!("unknown error: {:?}", e.type_id())
 }
+
+thread_local! {
+    static CURRENT_SPAN: Cell<Span> = Cell::new(DUMMY_SP);
+}
+
+pub struct CurrentSpanGuard {
+    old: Span,
+}
+
+impl Drop for CurrentSpanGuard {
+    fn drop(&mut self) {
+        CURRENT_SPAN.with(|cell| cell.set(self.old));
```

Comment:
am I interpreting this correctly as a funky RAII that replaces the old span when parent scope is entered?

Reply:
Yes, `set_current_span(x)` sets `CURRENT_SPAN` to `x`, and then when the guard goes out of scope, it resets `CURRENT_SPAN` to whatever value it held before (assuming you follow the normal pattern `let _g = set_current_span(x);` and don't move/swap/forget the guards).